libqrexec/exec.c

```diff
@@ -272,107 +303,89 @@ int execute_qubes_rpc_command(char *cmdline, int *pid, int *stdin_fd,
 static int execute_parsed_qubes_rpc_command(
         const struct qrexec_parsed_command *const command, int *const pid, int *const stdin_fd,
         int *const stdout_fd, int *const stderr_fd, struct buffer *stdin_buffer) {
-    char const *const delimiter = memchr(command->service_descriptor, '+', command->service_descriptor_length);
-    size_t const service_length = delimiter ?
-        (size_t)(delimiter - command->service_descriptor) : command->service_descriptor_length;
-
-    if (!service_length) {
+    if (!command->service_descriptor_length) {
         fputs("Service path empty\n", stderr);
         return -1;
-    } else if (service_length > NAME_MAX) {
-        fprintf(stderr, "Service path too long to execute: %zu\n", service_length);
+    } else if (command->service_descriptor_length > NAME_MAX) {
```

Comment:
This will break services where service name is ok, but its argument is long.

Reply:
Right. I reverted that part.